server/interceptor/interceptor.go

```diff
@@ -249,9 +249,9 @@ type rpcInfo struct {
 	// auth, getTree and quota enable their corresponding interceptor logic.
 	auth, getTree, quota bool
 
-	readonly bool
-	treeID   int64
-	treeType trillian.TreeType
+	readonly  bool
+	treeID    int64
+	treeTypes []trillian.TreeType
```

Comment:
This could be a map I think.

Reply:
The lines below would become too verbose...

`info.treeTypes = map[trillian.TreeType]bool{trillian.TreeType_LOG: true, trillian.TreeType_PREORDERED_LOG: true}`

Reply:
Can leave it for now. I might want to make all this logic a state machine if it's going to get more complicated.